projects/plugins/jetpack/extensions/blocks/videopress/edit.js

```diff
@@ -539,21 +541,34 @@ const VideoPressEdit = CoreVideoEdit =>
 
 			const { html, scripts } = preview;
 
+			// Render logic note :
+			// We make sure to exclude the VpBlock from the tree on preview reload so the HTML gets reloaded
+			// as we may receive the exact same HTML after the preview is resolved.
+			// Eslint disable note :
 			// Disabled because the overlay div doesn't actually have a role or functionality
 			// as far as the user is concerned. We're just catching the first click so that
 			// the block can be selected without interacting with the embed preview that the overlay covers.
 			/* eslint-disable jsx-a11y/no-static-element-interactions */
 			return (
 				<Fragment>
 					{ blockSettings }
-					<VpBlock
-						{ ...this.props }
-						hideOverlay={ this.hideOverlay }
-						html={ html }
-						scripts={ scripts }
-						interactive={ interactive }
-						caption={ caption }
-					/>
+					{ ( isUploading || isFetchingVideo ) && (
```

Comment:
I'm not sure we actually end up here when uploading... I tried uploading a video from a "completed" video block and it loads the first loading block, not this one (I changed the labels to confirm which one showed) So it might help to simplify this and the below condition to just the `isFetchingVideo` condition.

Reply:
I also changed the condition in displayLoadingBlock to make sure it takes the other branch on video upload